packages/components/tests/integration/components/hds/code-block/index-test.js

```diff
@@ -33,6 +38,23 @@ module('Integration | Component | hds/code-block/index', function (hooks) {
       .containsText("console.log('Hello world');");
   });
 
+  // DYNAMIC CONTENT
+
+  test('it renders the passed in dynamic content', async function (assert) {
+    this.set('value', "console.log('Hello world');");
+    await render(hbs`
+      <Hds::CodeBlock @value={{this.value}} id="test-code-block" />
+    `);
+    assert
+      .dom('#test-code-block pre code')
+      .hasText("console.log('Hello world');");
+    this.set('value', "console.log('Lorem ipsum');");
+    await settled();
```

Comment:
I initially tried with `rerender`, but it's too 'quick' (in a sense that it doesn't wait for the whole promise to end) so I ended up using `settled`.